src/macros.jl

```diff
@@ -168,10 +174,78 @@ macro addConstraint(m, x, extra...)
               "       expr1 <= expr2\n" * "       expr1 >= expr2\n" *
               "       expr1 == expr2\n" * "       lb <= expr <= ub")
     end
-
     return assert_validmodel(m, getloopedcode(c, code, :(), idxvars, idxsets, idxpairs, :ConstraintRef))
 end
 
+macro LinearConstraint(x)
+    (x.head == :block) &&
+        error("Code block passed as constraint. Perhaps you meant to use @LinearConstraints instead?")
+    (x.head != :comparison) &&
+        error("in @LinearConstraint ($(string(x))): expected comparison operator (<=, >=, or ==).")
+
+    if length(x.args) == 3
+        # Simple comparison - move everything to the LHS
+        if !((x.args[2] == :<=) || (x.args[2] == :≤) ||
+             (x.args[2] == :>=) || (x.args[2] == :≥) ||
+             (x.args[2] == :(==)))
+            error("in @LinearConstraint ($(string(x))): expected comparison operator (<=, >=, or ==).")
+        end
+        lhs = :($(x.args[1]) - $(x.args[3]))
+        newaff, parsecode = parseExpr(lhs, :q, [1.0])
```

Comment:
Just call ``@defExpr`` here. There's no need to interact with the low-level ``parseExpr`` code.